src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -208,109 +209,119 @@ public void render(Graphics g)
-      Graphics mapGraphics = mapImage.getGraphics();
-
-      // No overlay is being shown - draw the map, units, etc.
-      // Make sure the view is centered where we want it.
-      adjustViewLocation();
-
-      // Draw the portion of the base terrain that is currently in-window.
-      int drawMultiplier = SpriteLibrary.baseSpriteSize * SpriteOptions.getDrawScale();
-      int drawX = (int) (mapViewDrawX.get() * drawMultiplier);
-      int drawY = (int) (mapViewDrawY.get() * drawMultiplier);
-
-      // Make sure we specify draw coordinates that are valid per the underlying map image.
-      int maxDrawX = mapImage.getWidth() - mapViewWidth;
-      int maxDrawY = mapImage.getHeight() - mapViewHeight;
-      if( drawX > maxDrawX )
-        drawX = maxDrawX;
-      if( drawX < 0 )
-        drawX = 0;
-      if( drawY > maxDrawY )
-        drawY = maxDrawY;
-      if( drawY < 0 )
-        drawY = 0;
-
-      mapArtist.drawBaseTerrain(mapGraphics, gameMap, drawX, drawY, mapViewWidth, mapViewHeight);
-
-      // Update the central sprite indices so animations happen in sync.
-      updateAnimationIndices();
-
-      // Draw units, buildings, trees, etc.
-      drawUnitsAndMapObjects(mapGraphics, gameMap);
-
-      // Apply any relevant map highlight.
-      mapArtist.drawHighlights(mapGraphics);
-
-      // Draw Unit icons on top of everything, to make sure they are seen clearly.
-      drawUnitIcons(mapGraphics, gameMap);
-
-      // Get a reference to the current action being built, if one exists.
-      Unit currentActor = mapController.getContemplatedActor();
-      Path currentPath = mapController.getContemplatedMove();
-      boolean isTargeting = mapController.isTargeting();
-
-      // Draw the movement arrow if the user is contemplating a move/action (but not once the action commences).
-      if( null != currentPath && null == currentAnimation )
-      {
-        mapArtist.drawMovePath(mapGraphics, mapController.getContemplatedMove());
-      }
+    // We draw in three stages. First, we draw the map/units onto a canvas which is the size
+    // of the entire map; then we copy the visible section of that canvas onto a screen-sized
+    // image, then draw the overlay and scale that composite as we draw it to the window.
+    Graphics mapGraphics = mapImage.getGraphics();
+
+    // No overlay is being shown - draw the map, units, etc.
+    // Make sure the view is centered where we want it.
+    adjustViewLocation();
+
+    // Draw the portion of the base terrain that is currently in-window.
+    int drawMultiplier = SpriteLibrary.baseSpriteSize;
+    int drawX = (int) (mapViewDrawX.get() * drawMultiplier);
+    int drawY = (int) (mapViewDrawY.get() * drawMultiplier);
+
+    // Make sure we specify draw coordinates that are valid per the underlying map image.
+    int myWidth = mapViewWidth;
+    int myHeight = mapViewHeight;
+    int maxDrawX = mapImage.getWidth() - myWidth;
+    int maxDrawY = mapImage.getHeight() - myHeight;
```

Comment:
Why'd you add two new variables here?

Reply:
Good question. I might have gone through an intermediate step where it made sense?